### ISSUE TITLE
f5_cloud_failover_nic_map

### ISSUE THREAD

**Issue by jmcalalang** [F5Networks/f5-azure-arm-templates#172]
## Description
What is this tag used for: f5_cloud_failover_nic_map

From the CFE documentation:

> NIC mapping tag: the name is static but the value is user-provided (f5_cloud_failover_nic_map:<your value>) and must match the corresponding NIC on the secondary BIG-IP. The example below uses f5_cloud_failover_nic_map:external. This name/value tag will correspond to the name/value tag you use in the failoverAddresses.scopingTags section of the CFE declaration.

https://clouddocs.f5.com/products/extensions/f5-cloud-failover/latest/userguide/azure.html

However, after the deployment, if you query CFE that tag isn't even being listed:

```{
    "message": "success",
    "declaration": {
        "class": "Cloud_Failover",
        "environment": "azure",
        "externalStorage": {
            "scopingTags": {
                "f5_cloud_failover_label": "calalangKurath"
            }
        },
        "failoverAddresses": {
            "scopingTags": {
                "f5_cloud_failover_label": "calalangKurath"
            }
        },
        "failoverRoutes": {
            "scopingTags": {
                "f5_cloud_failover_label": "calalangKurath"
            },
            "scopingAddressRanges": [
                {
                    "range": "10.0.2.0/24"
                }
            ],
            "defaultNextHopAddresses": {
                "discoveryType": "static",
                "items": [
                    "10.0.2.4",
                    "10.0.2.5"
                ]
            }
        },
        "schemaVersion": "1.1.0"
    }
}```

This seems like either the ARM template is passing the tags wrong, or we're not adhering to the CFE standard use of tags.

## Template
https://github.com/F5Networks/f5-azure-arm-templates/tree/master/supported/failover/same-net/via-api/n-nic/new-stack/payg


## Severity Level
5

**Comment by JeffGiroux** [comment]
The ARM templates as of latest version use CFE and properly set all tags. The NIC tag is to correctly move the secondary IPs associated with that external or internal NIC to the corresponding BIG-IP #2 on external or internal NIC (respectively).

You can see the various tags for internal and external NICs.
https://github.com/F5Networks/f5-azure-arm-templates/blob/2fac82ec2520a8644b0d9e820ab511e6b1237cc2/supported/failover/same-net/via-api/n-nic/new-stack/payg/azuredeploy.json#L926

I asked PM for more clarification on this. Docs should be updated with other tagging clarification for NIC mapping in upcoming release.

**Comment by jmcalalang** [comment]
so thats the assumption, but why is it not in CFE when you query it?

**Comment by shyawnkarim** [comment]
Thanks for providing this feedback about the f5_cloud_failover_nic_map tag documentation. I've created internal ID AUTOSDK-257 to get this section of our documentation fixed.

The f5_cloud_failover_nic_map tag is not part of the declaration, which is why you are not able to see it.

**Comment by jmcalalang** [comment]
We could use some explanation on that @shyawnkarim, since the doc says **f5_cloud_failover_nic_map** should be in it :).

**Comment by vvt137** [comment]
Have there been any updates regarding this ticket? It has remained opened for 4 months and there were 5 (!) new releases of CFE documentation since then... Should we, perhaps, open a separate issue ticket in the CFE repo?

Note that the AWS templates such as https://github.com/F5Networks/f5-aws-cloudformation/tree/master/supported/failover/across-net/via-api/3nic/existing-stack/payg still do not have NICs tagged with f5_cloud_failover_nic_map. CEF appears to operate well, but the impact of not having these tags is unknown.

**Comment by shyawnkarim** [comment]
Closing.

The documentation was corrected in [Cloud Failover Extension Release 1.4](https://github.com/F5Networks/f5-cloud-failover-extension/releases/tag/v1.4.0).
